posthog/management/commands/test/test_fix_person_distinct_ids_after_delete.py

```diff
@@ -0,0 +1,187 @@
+import logging
+from unittest import mock
+from uuid import UUID, uuid4
+
+import pytest
+
+import posthog.management.commands.fix_person_distinct_ids_after_delete
+from posthog.client import sync_execute
+from posthog.management.commands.fix_person_distinct_ids_after_delete import run
+from posthog.models.person.person import Person, PersonDistinctId
+from posthog.models.person.sql import PERSON_DISTINCT_ID2_TABLE
+from posthog.models.person.util import create_person, create_person_distinct_id
+from posthog.test.base import BaseTest, ClickhouseTestMixin
+
+
+@pytest.mark.ee
+class TestFixPersonDistinctIdsAfterDelete(BaseTest, ClickhouseTestMixin):
+    CLASS_DATA_LEVEL_SETUP = False
+
+    @mock.patch(
+        f"{posthog.management.commands.fix_person_distinct_ids_after_delete.__name__}.create_person_distinct_id",
+        wraps=posthog.management.commands.fix_person_distinct_ids_after_delete.create_person_distinct_id,
+    )
+    def test_dry_run(self, mocked_ch_call):
+        # clickhouse only deleted person and distinct id that should be updated
+        ch_only_deleted_person_uuid = create_person(
+            uuid=str(uuid4()), team_id=self.team.pk, is_deleted=True, version=5, sync=True
+        )
+        create_person_distinct_id(
+            team_id=self.team.pk,
+            distinct_id="distinct_id",
+            person_id=ch_only_deleted_person_uuid,
+            is_deleted=True,
+            version=7,
+            sync=True,
+        )
+        # reuse
+        person_linked_to_after = Person.objects.create(
+            team_id=self.team.pk, properties={"abcdefg": 11112}, version=1, uuid=uuid4()
+        )
+        PersonDistinctId.objects.create(
+            team=self.team, person=person_linked_to_after, distinct_id="distinct_id", version=0
+        )
+        options = {"live_run": False, "team_id": self.team.pk, "new_version": 2500}
+        run(options, True)
+
+        # postgres didn't change
+        pg_distinct_ids = PersonDistinctId.objects.all()
+        self.assertEqual(len(pg_distinct_ids), 1)
+        self.assertEqual(pg_distinct_ids[0].version, 0)
+        self.assertEqual(pg_distinct_ids[0].distinct_id, "distinct_id")
+        self.assertEqual(pg_distinct_ids[0].person.uuid, person_linked_to_after.uuid)
+
+        # CH didn't change
+        ch_person_distinct_ids = sync_execute(
+            f"""
+            SELECT person_id, team_id, distinct_id, version, is_deleted FROM {PERSON_DISTINCT_ID2_TABLE} FINAL WHERE team_id = %(team_id)s ORDER BY version
+            """,
+            {"team_id": self.team.pk},
+        )
+        self.assertEqual(
+            ch_person_distinct_ids,
+            [
+                (UUID(ch_only_deleted_person_uuid), self.team.pk, "distinct_id", 7, True),
+            ],
+        )
+        mocked_ch_call.assert_not_called()
```

Comment:
I don't use mocks much but I've seen it being mentioned to use an actually assert rather than calling methods on mocks, as they could silently fail.

Reply:
Or pass even